ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/TestGeneratedDefaultConfigIsUpToDate.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.sandbox
+
+import java.nio.charset.StandardCharsets
+
+import org.apache.commons.io.IOUtils
+import org.scalatest.{AppendedClues, OptionValues}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class TestGeneratedDefaultConfigIsUpToDate
+    extends AnyFlatSpec
+    with Matchers
+    with OptionValues
+    with AppendedClues {
+
+  private val clue =
+    "Generated default config is not up-to-date. \nPlease run: bazel run //ledger/sandbox-on-x:update-generated-sources"
+
+  it should "generated default config should be up-to-date" in {
```

Comment:
I am not sure what we are trying to achieve in a big picture with this. Let's say config changed and I as a developer forgot to regenerate the default config file. What am I gaining by knowing that I did?

Reply:
OK, got it.
Note that we also leave some of the fields deliberately unset in the `reference.conf`.